packages/maskbook/src/extension/background-script/EthereumServices/providers/MetaMask.ts

```diff
@@ -63,6 +64,11 @@ export async function requestAccounts() {
     return accounts
 }
 
+export async function checkIfMetaMaskLocked() {
+    const accounts: string[] = await provider!.request({ method: 'eth_requestAccounts', params: [] })
```

Comment:
I think that also applies if there is really no accounts in MetaMask, or user doesn't grant MetaMask wallet access to Maskbook

Reply:
yeah, here requires more consideration.

Reply:
FYI: https://docs.metamask.io/guide/ethereum-provider.html#ethereum-metamask-isunlocked

Does this API works?

Reply:
Thanks! It works on the new version of @metamask/inpage-provider, The old `metamask-extension-provider` does not support it as well.